metadata/metadata.go

```diff
@@ -257,6 +294,12 @@ func (sm StagedMetadata) IsDefault() bool {
 	return sm.CutoverNanos == 0 && !sm.Tombstoned && sm.Metadata.IsDefault()
 }
 
+// IsDropPolicyApplied returns whether this is the default staged metadata
```

Comment:
I see, I guess my main question is that, when `Drop` is being applied, do we want to enforce all other metadatas are set to `default`?

Reply:
I'd feel that's probably best, otherwise there is a mangled rule.  It should be drop set and nothing else.

Reply:
I guess that means when there is a non default storage policy or aggID defined in the pipeline it will override the `Drop` policy, we should make sure we have the validation logic to check and prevent this kind of rules @jskelcy